public/app/features/admin/UserListAdminPage.tsx

```diff
@@ -0,0 +1,141 @@
+              tabIndex={1}
+              autoFocus={true}
+              value={props.query}
+              spellCheck={false}
+              onChange={event => props.changeQuery(event.currentTarget.value)}
+              prefix={<i className="fa fa-search" />}
+            />
+            <ActionBarSpacing />
+            <Forms.LinkButton href="admin/users/create" variant="primary">
+              New user
+            </Forms.LinkButton>
+          </ActionBar>
+
+          <div className="admin-list-table">
+            <table className="filter-table form-inline filter-table--hover">
+              <thead>
+                <tr>
+                  <th></th>
+                  <th>Login</th>
+                  <th>Email</th>
+                  <th>
+                    Seen&nbsp;
+                    <Tooltip placement="top" content="Time since user was seen using Grafana">
+                      <i className="fa fa-question-circle" />
+                    </Tooltip>
+                  </th>
+                  <th></th>
+                  <th style={{ width: '1%' }}></th>
+                </tr>
+              </thead>
+              <tbody>{props.users.map(renderUser)}</tbody>
+            </table>
+          </div>
+          <Pagination
+            visible={props.showPaging}
+            numberOfPages={props.totalPages}
+            currentPage={props.page}
+            onNavigate={props.changePage}
+          />
+        </>
+      </Page.Contents>
+    </Page>
+  );
+};
+
+const renderUser = (user: UserDTO) => {
+  const editUrl = `admin/users/edit/${user.id}`;
+
+  return (
+    <tr key={user.id}>
+      <td className="width-4 text-center link-td">
+        <a href={editUrl}>
+          <img className="filter-table__avatar" src={user.avatarUrl} />
+        </a>
+      </td>
+      <td className="link-td">
+        <a href={editUrl}>{user.login}</a>
+      </td>
+      <td className="link-td">
+        <a href={editUrl}>{user.email}</a>
+      </td>
+      <td className="link-td">{user.lastSeenAtAge && <a href={editUrl}>{user.lastSeenAtAge}</a>}</td>
+      <td className="link-td">
+        {user.isAdmin && (
+          <a href={editUrl}>
+            <i className="fa fa-shield" bs-tooltip="'Grafana Admin'" />
+          </a>
+        )}
+      </td>
+      <td className="text-right">
+        {Array.isArray(user.authLabels) && user.authLabels.length > 0 && (
+          <TagBadge label={user.authLabels[0]} removeIcon={false} count={0} onClick={() => {}} />
+        )}
+      </td>
+      <td className="text-right">
+        {user.isDisabled && <span className="label label-tag label-tag--gray">Disabled</span>}
+      </td>
+    </tr>
+  );
+};
+
+const mapDispatchToProps: MapDispatchToProps<DispatchProps, OwnProps> = {
+  fetchUsers,
+  changeQuery,
+  changePage,
+};
+
+const mapStateToProps: MapStateToProps<ConnectedProps, OwnProps, StoreState> = state => ({
+  navModel: getNavModel(state.navIndex, 'global-users'),
```

Comment:
This may have a performance penalty - looking at the source code it looks like it'd re-render the component on every state change, since `getNavModel` is not optimised to cache the previous result. Instead, it would recalculate the result no matter wether `state.navIndex` changed or not.

Possible solution: use reselect. It does a great job avoiding re-calculations. Reselect is a dependency anyway.

Reply:
I see your point but and love the suggestions! But as far as I know we don't have any issues with performance and might not need to optimize on this low level (yet). 

Regarding reselect, have been working with that before, it adds a layer of complexity and I would only add it if really needed.

Reply:
Agree with @mckn